Change granularity of time to 0.5 days

punchagan opened the pull request:
The parser has been modified to support times only at the granularity of 0.5 days. We check if the time value is .5 or a valid number with an optional decimal point followed by an optional 0 or 5.

Closes #156

Co-authored-by: Ganesh Narayanan

jmid commented on test/cram/lint/engineer.t:
```diff
@@ -87,3 +91,51 @@ Only "No KR" and "New KR" are supported for KR's without identifiers
+  [ERROR(S)]: <stdin>
+  
+  In KR "@eng1 (1.1 day)":
+    Invalid time entry found. Format is '- @eng1 (x days), @eng2 (x days)'
```
I suggest we improve the error message here.
Just stating that it is invalid gives little clue as to what the problem is with filing, e.g., 1.75 days.
A suggestion:
```
   Invalid time entry found. 
   Format is '- @eng1 (x days), @eng2 (y days)' where x and y must be divisible by 0.5
```

ganeshn-gh replied:
Hey @jmid, thanks for your suggestion. We have implemented it.

gpetiot reviewed (approved):
Looks good, thanks!

I just promoted the tests, if you're done with it we can merge.

punchagan commented:
> I just promoted the tests, if you're done with it we can merge.

Thanks! We're done with it; please merge! (I squashed your commit into the first commit, with you as a co-author)

jmid commented:
I think something went wrong with the last force push, so that the updated error message in the cram tests (*.t) no longer shows the revised error message.

Another nit: be mindful of overflowing a 80 column width with the error message, as it makes it substantially harder for @shindere to read it. For this reason, I was suggesting a newline after `Invalid time entry found.`.

punchagan commented:
> I think something went wrong with the last force push, so that the updated error message in the cram tests (*.t) no longer shows the revised error message.

Thanks for catching this! I've fixed it. 



> Another nit: be mindful of overflowing a 80 column width with the error message, as it makes it substantially harder for @shindere to read it. For this reason, I was suggesting a newline after `Invalid time entry found.`.

Thanks for being mindful of this! <3

I've added a couple of break hints which formats the error message better for 80 column wide terminals.

jmid commented on lib/lint.ml:
```diff
@@ -59,8 +59,9 @@ let pp_error ppf = function
-         days), @@eng2 (x days)'@]@,"
+        "@[<hv 2>In KR %S:@ Invalid time entry found.@,\
+        \ Format is '- @@eng1 (x days), @@eng2 (y days)@,\
+        \ where x and y must be divisible by 0.5'@]@,"
```
Thanks, this is shaping up nicely!
I just spotted that the second quote is off: It should read: `Format is '- @@eng1 (x days), @@eng2 (y days)'` (notice the final quote character `'`).

Very minor nit: 3 lines for the error message is perhaps a bit too much?
If we keep the first two on the same line, the result is still only 74 chars wide:
```
    Invalid time entry found. Format is '- @eng1 (x days), @eng2 (y days)'
     where x and y must be divisible by 0.5
```

gpetiot replied:
Done, I fixed the quote and made the message fit on 2 lines.

shindere commented:
Jan Midtgaard (2024/03/12 07:19 -0700):
> I think something went wrong with the last force push, so that the updated error message in the cram tests (*.t) no longer shows the revised error message.
>
> Another nit: be mindful of overflowing a 80 column width with the
> error message, as it makes it substantially harder for @shindere to
> read it. For this reason, I was suggesting a newline after `Invalid
> time entry found.`.

That would indeed be way better!

Many thanks for your attention to details Jan, vey much appreciated!

shindere commented:
Puneeth Chaganti (2024/03/12 07:47 -0700):
> > I think something went wrong with the last force push, so that the updated error message in the cram tests (*.t) no longer shows the revised error message.
>
> Thanks for catching this! I've fixed it.
>
>
>
> > Another nit: be mindful of overflowing a 80 column width with the error message, as it makes it substantially harder for @shindere to read it. For this reason, I was suggesting a newline after `Invalid time entry found.`.
>
> Thanks for being mindful of this! <3
>
> I've added a couple of break hints which formats the error message
> better for 80 column wide terminals.

Thank you!
